src/adapters/MeetingsJSONAdapter.test.js

```diff
@@ -1,57 +1,133 @@
+        .subscribe((meeting) => {
+          expect(meeting.localAudio.muted).toBeTruthy();
+          done();
+        });
+    });
+
+    test('throws a proper error message', (done) => {
+      const wrongMeetingID = 'wrongMeetingID';
+
+      subscription = meetingsJSONAdapter.getMeeting(wrongMeetingID).subscribe(
+        () => {},
+        (error) => {
+          expect(error.message).toBe(`Could not find meeting with ID "${wrongMeetingID}"`);
+          done();
+        }
+      );
+    });
   });
 
-  test('getMeeting() completes the observable', (done) => {
-    meetingsJSONAdapter.getMeeting(meetingID).subscribe(
-      () => {},
-      () => {},
-      () => {
-        expect(true).toBeTruthy();
-        done();
-      }
-    );
+  describe('muteAudioControl() returns', () => {
+    beforeEach(() => {
+      meetingID = 'localAudio';
+    });
+
+    test('active control display values', (done) => {
+      rxjs.fromEvent = jest.fn(() => from([{detail: meetings[meetingID]}]));
+
+      subscription = meetingsJSONAdapter
+        .muteAudioControl(meetingID)
+        .pipe(skip(1)) // Skip the "default" emission
+        .subscribe((display) => {
+          expect(display).toMatchObject({
+            ID: 'mute-audio',
+            icon: 'microphone-muted',
+            tooltip: 'Unmute',
+            state: 'active',
+          });
+          done();
+        });
+    });
+
+    test('inactive control display values', (done) => {
+      rxjs.fromEvent = jest.fn(() => {
+        const meeting = meetings[meetingID];
+
+        meeting.localAudio.muted = false; // Local audio is already muted
+
+        return from([{detail: meeting}]);
+      });
+
+      subscription = meetingsJSONAdapter
+        .muteAudioControl(meetingID)
+        .pipe(skip(1)) // Skip the "default" emission
+        .subscribe((display) => {
+          expect(display).toMatchObject({
+            ID: 'mute-audio',
+            icon: 'microphone',
+            tooltip: 'Mute',
+            state: 'inactive',
+          });
+          done();
+        });
+    });
+
+    test('error on invalid meeting ID', (done) => {
+      subscription = meetingsJSONAdapter.muteAudioControl('invalid').subscribe(
+        () => {},
+        (error) => {
+          expect(error.message).toEqual('Could not find meeting with ID "invalid"');
+          done();
+        }
+      );
+    });
   });
 
-  test('getMeeting() renders the local media if localVideo property is defined', (done) => {
-    meetingsJSONAdapter.getMeeting('localVideo').subscribe((data) => {
-      expect(data.localVideo).toEqual('mock-stream');
-      done();
+  describe('toggleMuteAudio()', () => {
+    test('dispatches a "mute-audio" event', () => {
+      meetingID = 'localAudio';
+
+      meetingsJSONAdapter.toggleMuteAudio(meetingID);
+      expect(global.document.dispatchEvent).toHaveBeenCalled();
     });
   });
 
   afterEach(() => {
+    if (subscription) {
```

Comment:
you are basically doing the check for two test cases out of however many we have total. subscription should be available throughout the whole suite, since it is utilized for most of the test cases. this is how we set variables globally in most our test suites. if there is an object being utilized across multiple test cases, mind as well create them at the top level for clarity and accessibility. 

not all these objects have been utilized in all test cases but they have beed defined at the top level:

```js
[meetingID] = Object.keys(meetings);
    meetingsJSONAdapter = new MeetingsJSONAdapter(meetings);	    meetingsJSONAdapter = new MeetingsJSONAdapter(meetings);
    meetingsJSONAdapter.getLocalVideo = jest.fn(() => Promise.resolve('mock-stream'));	    meetingsJSONAdapter.getLocalVideo = jest.fn(() => Promise.resolve('mock-stream'));
```
that we could remove this check statement too. 
also, extra subscription here means not much since this is all mock data.

Reply:
RESOLVED!!!!